python/strings_udf/strings_udf/__init__.py

```diff
@@ -41,59 +27,65 @@ def _get_appropriate_file(sms, cc):
         return None
 
 
+def maybe_patch_numba_linker(driver_version):
+    # Numba thinks cubinlinker is only needed if the driver is older than the ctk
+    # but when strings_udf is present, it might also need to patch because the PTX
+    # file strings_udf relies on may be newer than the driver as well
+    if driver_version < strings_udf_ptx_version:
+        logger.debug(
+            "Driver version %s.%s needs patching due to strings_udf"
+            % driver_version
+        )
+        if _numba_version_ok:
+            logger.debug("Patching Numba Linker")
+            Linker.new = new_patched_linker
+        else:
+            logger.debug("Cannot patch Numba Linker - unsupported version")
+
+
 # adapted from PTXCompiler
 cp = subprocess.run([sys.executable, "-c", CMD], capture_output=True)
+# must have a driver to proceed
```

Comment:
I would like to be able to figure out a way to handle this such that the runtime subprocess forking is optional (see https://github.com/rapidsai/ptxcompiler/pull/20 for context); @gmarkall did you do a similar thing with cubinlinker as we did for ptxcompiler? See also the docs for dask-cuda here which explain this in a bit more detail: https://docs.rapids.ai/api/dask-cuda/stable/ucx.html#running-in-a-fork-starved-environment

At the very least, we should try and avoid executing this subprocess fork at import time and delay until the resulting `ptxpath` variable is actually needed (although I guess you're patching `cudf/udf/utils.py` at import time with the result of this call).